### PR TITLE
print header and print_trailer

### PR DESCRIPTION
**Pull request by bbannon**
When calling print_header with parameters, always make $includes an array; even if it's empty.
When the parameters in either are just tested for truthy, use 0 or 1.
Where ever print_header and ob_start were called together,
or print_trailer and ob_end_flush were called together,
move those calls into the functions.

Clean up a couple of places where js files were called in $headx.
Clean up a couple of places where things were called incorrectly.

### CODE REVIEW

**Comment by bbannon** [comment]
If the conflict is the ckeditor version, use the latest.

File: availability.php [hunk 1]
```diff
@@ -76,6 +76,6 @@
     <form action="availability.php" method="post">
       ' . daily_matrix ( $date, $users ) . '
     </form>
-    ' . print_trailer ( false, true, true );
+    ' . print_trailer ( 0, 1, 1 );
```
Comment (craigk5n) [inline availability.php]: What is the benefit of change 0 to false here?

File: activity_log.php [hunk 2]
```diff
@@ -30,9 +30,6 @@
 $sys = ( $is_admin && getGetValue ( 'system' ) != '' );
 
 print_header();
-
-ob_start();
```
Comment (craigk5n) [inline activity_log.php]: Can you recall why all these ob_start calls are in the code?  It seems unnecessary, but they must have been put there for a reason(?).

File: availability.php (hunk 1 above)
Comment (bbannon) [inline availability.php]: That one line is 10 characters shorter. There are hundreds of such lines, not just for print_trailer but other functions. Since most of the functions only test for truthy anyway, and very few functions test for '=== false', we could probably cut several seconds off load times when/if we convert all of them. Even on my DSL line, it takes too long to load as is.
And we could probably save thousands of hours of typing in future. :)

File: activity_log.php (hunk 2 above)
Comment (bbannon) [inline activity_log.php]: I put them in back when I only had a dial up modem. Sending pages in bigger chunks was WAY more efficient. However, I'm taking most of them out now.
Also, there used to be places where characters were sent and then headers/cookies were tried to send. ob_start() sorted those out. I think that problem has been resolved now, too.

File: availability.php (hunk 1 above)
Comment (craigk5n) [inline availability.php]: Using "false" provides more context to the developer than "0".  The 0 and false should be identical to the end user, right?  It might be a tiny bit slower to parse on anything earlier than PHP 5.5.  According this [this](https://www.acquia.com/blog/php-getting-faster), PHP 5.5 and later essentially compile and cache the compiled PHP.

I'm okay with larger source files (adding comments, etc.) if it provides better information to other developers.

Comment (bbannon) [inline availability.php]: Well, I left the default parameters as true/false in the functions.
Since we don't send in named parameters, developers have to go look at the function itself to see what's going on anyway.
Once I saw;
function x ($a=false, $b=true) {}
I pretty much knew what incoming 0 and 1 were doing everywhere.

If you really want to keep true/false, I'll stop converting them.

File: activity_log.php (hunk 2 above)
Comment (bbannon) [inline activity_log.php]: "This branch has conflicts"
Damn. I've managed to corrupt the master on my fork.
The "fixes" I've tried just make it worse.
Is there a way to "re-fork" or re sync my master with yours?
Without creating a new account?

File: availability.php (hunk 1 above)
Comment (bbannon) [inline availability.php]: I didn't say that very well. Let me try again.
print_trailer(true) or print_trailer(1)
Neither tells me anything. There is no context. All I know is, we call a function that takes one parameter that most likely tests for truthy.
I have to go look at function print_trailer($include_nav_links) to even see what the variable is named.
I would rather cut a few thousand K bytes.
Well, actually, I would rather not put them to start with. Oh, well.

Comment (craigk5n) [inline availability.php]: Hey, Bruce.  I think we're on the same page here mostly.  I'm hesitant about changes to large numbers of files because it's time-consuming to test and validate (especially without any unit tests or CI in place).

File: activity_log.php (hunk 2 above)
Comment (craigk5n) [inline activity_log.php]: You can just create a new working directory someplace else by starting a new clone.

Comment (bbannon) [inline activity_log.php]: I meant, I screwed up the master on my github copy.
git commit and push there.
I tried cloning from your original and my fork. I couldn't figure out how to change "yours" to mine on my computer. And mine is toast.
I imagine that github newbies do this a lot. It's really easy to mess it up. There must be a way to undo or re-sync. At least, I hope there is.

File: availability.php (hunk 1 above)
Comment (bbannon) [inline availability.php]: OK. "Not changing large numbers of files at once" makes more sense. I've been doing global search and individual replace. I do look over the diff file at least six or seven times over a couple of days. And I call up the program in WAMP to look it over. Though I need to upgrade to PHP7 on that. Maybe it would better catch the occasional extra quote, etc.

File: activity_log.php (hunk 2 above)
Comment (craigk5n) [inline activity_log.php]: You can use various forms of 'git reset' to alter your local git repo.  Lots of good examples on stackexchange you can find with google.  And, yes, lots of people run into issues like this.

File: availability.php (hunk 1 above)
Comment (craigk5n) [inline availability.php]: You can try out PHP 7.1 with the a docker container: https://github.com/craigk5n/webcalendar-docker

That's how I've been testing PHP 7. I generally just use "php -l" on all the php files. But I'd love to use Selenium IDE to create some automated tests... just haven't got around to it.

File: activity_log.php (hunk 2 above)
Comment (bbannon) [inline activity_log.php]: Great! It looks like I need to do git reset --hard <commit>. But, how do I know when your master and mine are in sync again? Especially since you've updated your master since I screwed up mine. Do I use '55d1ec6', which is the last commit you made? The manual is not real clear on that.

Comment (craigk5n) [inline activity_log.php]: Look for variations of the 'git status' command

**Comment by craigk5n** [comment]
Closing due to merge conflicts.  If conflicts are resolved, please re-open.
